From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: NONPLAYT <76615486+NONPLAYT@users.noreply.github.com>
Date: Mon, 7 Jul 2025 04:06:32 +0300
Subject: [PATCH] Replace CraftServer#getWorlds ArrayList with a fastutil
 ObjectArrayList

CraftServer#getWorlds is frequently used in some plugins, so replacing ArrayList with fastutil ObjectArrayList will give some performance boost

diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 17022a57cb06b0ddd0df94075c0d92b68137d901..414fdc6ca7a2979124a7fbf529bff60f993c005a 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -974,7 +974,7 @@ public final class CraftServer implements Server {
 
     @Override
     public List<World> getWorlds() {
-        return new ArrayList<World>(this.worlds.values());
+        return new it.unimi.dsi.fastutil.objects.ObjectArrayList<World>(this.worlds.values()); // DivineMC - Replace CraftServer#getWorlds ArrayList with a fastutil ObjectArrayList
     }
 
     @Override
